src/main/java/uk/nhs/digital/nhsconnect/nhais/translator/amendment/AmendmentNameToEdifactMapper.java

```diff
@@ -0,0 +1,100 @@
+package uk.nhs.digital.nhsconnect.nhais.translator.amendment;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.FhirValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.PersonName;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Segment;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class AmendmentNameToEdifactMapper extends AmendmentToEdifactMapper {
+    @Override
+    protected Optional<Segment> mapPatches(JsonPatches patches) {
+        var title = patches.getTitle()
+            .map(this::getValue)
+            .orElse(null);
+        var surname = patches.getSurname()
+            .map(this::getValue)
+            .orElse(null);
+        var firstName = patches.getFirstForename()
+            .or(() -> patches
+                .getAllForenamesPath()
+                .filter(patch -> patch.getOp() == AmendmentPatchOperation.REMOVE))
+            .map(this::getValue)
+            .orElse(null);
+        var secondName = patches.getSecondForename()
+            .map(this::getValue)
+            .orElse(null);
+        var otherName = patches.getOtherForenames()
+            .map(this::getValue)
+            .orElse(null);
+
+        var personName = PersonName.builder()
+            .nhsNumber(patches.getAmendmentBody().getNhsNumber())
```

Comment:
Maybe we should pass `AmendmentBody` to this method instead of making cyclic dependencies? Or just add NhsNumber to `JsonPatches`